packages/safe-apps-sdk/src/safe/index.ts

```diff
@@ -28,6 +39,100 @@ class Safe {
 
     return response.data;
   }
+
+  calculateMessageHash(message: BytesLike): string {
+    if (typeof message === 'string') {
+      message = ethers.utils.toUtf8Bytes(message);
+    }
+    return ethers.utils.keccak256(message);
+  }
+
+  private async check1271Signature(messageHash: Uint8Array, signature = '0x'): Promise<boolean> {
+    const safeInfo = await this.getInfo();
+
+    const encodedIsValidSignatureCall = EIP_1271_INTERFACE.encodeFunctionData('isValidSignature', [
+      messageHash,
+      signature,
+    ]);
+
+    const payload = {
+      call: RPC_CALLS.eth_call,
+      params: [
+        {
+          to: safeInfo.safeAddress,
+          data: encodedIsValidSignatureCall,
+        },
+        'latest',
+      ],
+    };
+    try {
+      const response = await this.communicator.send<Methods.rpcCall, RPCPayload<[TransactionConfig, string]>, string>(
+        Methods.rpcCall,
+        payload,
+      );
+
+      return response.data.slice(0, 10).toLowerCase() === MAGIC_VALUE;
+    } catch (err) {
+      return false;
+    }
+  }
+
+  private async check1271SignatureBytes(messageHash: Uint8Array, signature = '0x'): Promise<boolean> {
+    const safeInfo = await this.getInfo();
+
+    const encodedIsValidSignatureCall = EIP_1271_BYTES_INTERFACE.encodeFunctionData('isValidSignature', [
+      messageHash,
+      signature,
+    ]);
+
+    const payload = {
+      call: RPC_CALLS.eth_call,
+      params: [
+        {
+          to: safeInfo.safeAddress,
+          data: encodedIsValidSignatureCall,
+        },
+        'latest',
+      ],
+    };
+
+    try {
+      const response = await this.communicator.send<Methods.rpcCall, RPCPayload<[TransactionConfig, string]>, string>(
+        Methods.rpcCall,
+        payload,
+      );
+
+      return response.data.slice(0, 10).toLowerCase() === MAGIC_VALUE_BYTES;
+    } catch (err) {
+      return false;
+    }
+  }
+
+  async isMessageSigned(message: BytesLike, signature = '0x'): Promise<boolean> {
```

Comment:
Is there a limitation on the message? (e.g. that it is 32 bytes long)

Reply:
no, should there be one? I took the validation logic from https://github.com/authereum/is-valid-signature

Reply:
Does it make sense to rename this message to `isMessageHashSigned` as this would make it more clear that you should hash the message before checking it. And maybe we have another method `isMessageSigned` which hashes it before.

Reply:
@rmeissner Could you please take a quick look at this? https://github.com/gnosis/safe-apps-sdk/pull/211/commits/ff0c7467966f0f00b3559a654dd0a380cc28a187